### PR TITLE
refs #100 fixes tox lint

### PR DESCRIPTION
**Pull request by johannesjh**
* does not specify a different beancount version for the tox lint environment
* disables specific pylint warnings

### CODE REVIEW

File: smart_importer/entries.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ def merge_non_transaction_entries(imported_entries, enhanced_transactions):
     enhanced_entries = []
     enhanced_transactions_iter = iter(enhanced_transactions)
     for entry in imported_entries:
-        if isinstance(entry, Transaction):
+        if isinstance(entry, Transaction):  # pylint: disable=W1116
```
Comment (yagebu) [inline smart_importer/entries.py:44]:
```suggestion
        if isinstance(entry, Transaction):  # pylint: disable=isinstance-second-argument-not-valid-type
```

**Review by yagebu — CHANGES_REQUESTED** [review]
I think using the full pylint warning names is preferable. I this makes the lines too long, you could also add the disable at the block level (just below `for entry in imported_entries`)

**Comment by johannesjh** [comment]
@yagebu The sourcehut build fails with error message "No module named 'setuptools'". That is rather strange, setuptools should be part of the python installation, right?

**Comment by yagebu** [comment]
> The sourcehut build fails with error message "No module named 'setuptools'". That is rather strange, setuptools should be part of the python installation, right?

It is not part of the Python distribution, so we need to install it separately. See #102.

We do specify it in our pyproject.toml, so theoretically it should be pulled in automatically, but that doesn't seem to happen.

**Comment by yagebu** [comment]
@johannesjh: It actually needs both pip and setuptools. I've pushed the fixes to the Arch CI to master, so you can just rebase this branch

File: smart_importer/entries.py (hunk 1 above)
Comment (johannesjh) [reply smart_importer/entries.py]: done, thank you for the suggestion

**Comment by johannesjh** [comment]
We really should freeze package versions. Pylint on arch linux now complains about issues that were ignored on ubuntu.

**Comment by yagebu** [comment]
> We really should freeze package versions.

I'm not a fan of freezing package versions without an automatic process in place that updates them. I think bumping the minimum version would be preferable.

**Comment by johannesjh** [comment]
Can you reproduce the unsubscriptable-object warning? 
I never got this warning. I tried specifying `pylint>=2.6.2` in tox.ini, I still do not get the warning.

It might have to do with python 3.9, see
https://github.com/PyCQA/pylint/issues/3882

**Comment by yagebu** [comment]
> It might have to do with python 3.9

Yes, I've seen this one before, this is not due to the pylint version but due to the Python version

**Comment by johannesjh** [comment]
I added a github ci build matrix that reproduces the issue. As a workaround, I will instruct pylint to ignore these warnings.

**Comment by johannesjh** [comment]
Thank you for your help. I think we are all green now.

**Review by yagebu — APPROVED** [review]
LGTM. I've just moved two of the pylint disables.

**Pull request merged**